File: Lua/Framework/Manager/Wrapper/VideoClipManager.lua
local BaseManager = require("BaseManager")

---@class VideoClipManager:BaseManager
---@field new fun():VideoClipManager
---@field super BaseManager
local VideoClipManager = class('VideoClipManager', BaseManager)

function VideoClipManager:ctor()
    ---@type CS.DragonReborn.AssetTool.VideoClipManager
    self.manager = CS.DragonReborn.AssetTool.VideoClipManager.Instance
    self.manager:OnGameInitialize(nil)
end

function VideoClipManager:Reset()
    try_catch_traceback_with_vararg(self.manager.Reset, nil, self.manager)
end

function VideoClipManager:OnLowMemory()
    try_catch_traceback_with_vararg(self.manager.OnLowMemory, nil, self.manager)
end

function VideoClipManager:LoadVideoClip(assetName)
    return self.manager:LoadVideoClip(assetName)
end

function VideoClipManager:UnloadVideoClip(assetName)
    self.manager:UnloadVideoClip(assetName)
end

return VideoClipManager

File: Packages/com.funplus.dragonreborn.assettool@0.0.1/Runtime/AssetCreate/Video/VideoClipManager.cs
using System.Collections.Generic;
using UnityEngine.Video;

// ReSharper disable once CheckNamespace
namespace DragonReborn.AssetTool
{
	public class VideoClipManager : Singleton<VideoClipManager>, IManager
	{
		private Dictionary<string, AssetHandle> _allLoadedVideClip;

		public void OnGameInitialize(object configParam)
		{
			_allLoadedVideClip = new Dictionary<string, AssetHandle>();
		}

		public void Reset()
		{
			if (null == _allLoadedVideClip) return;
			foreach (var handle in _allLoadedVideClip.Values)
			{
				AssetManager.Instance.UnloadAsset(handle);
			}
			_allLoadedVideClip.Clear();
		}

		public VideoClip LoadVideoClip(string assetName)
		{
			if (null != _allLoadedVideClip && _allLoadedVideClip.TryGetValue(assetName, out var handle))
			{
				return handle.Asset as VideoClip;
			}

			// video clip 只能从resource 无压缩方式加载
			handle = AssetManager.Instance.LoadAsset(assetName, false, AssetManager.SyncLoadReason.ResourceLoad);
			if (handle.IsValid)
			{
				_allLoadedVideClip?.Add(assetName, handle);
				return handle.Asset as VideoClip;
			}
			NLogger.Error($"LoadVideoClip {assetName} failed.");
			return null;
		}

		public void UnloadVideoClip(string assetName)
		{
			if (null == _allLoadedVideClip || !_allLoadedVideClip.TryGetValue(assetName, out var handle)) return;
			_allLoadedVideClip.Remove(assetName);
			AssetManager.Instance.UnloadAsset(handle);
		}

		public void OnLowMemory()
		{

		}
	}
}
